parquet-column/src/main/java/org/apache/parquet/column/statistics/Statistics.java

```diff
@@ -175,9 +217,48 @@ public void mergeStatistics(Statistics stats) {
    */
   abstract public void setMinMaxFromBytes(byte[] minBytes, byte[] maxBytes);
 
+  /**
+   * Returns the min value in the statistics. The java natural order of the returned type defined by {@link
+   * T#compareTo(Object)} might not be the proper one. For example, UINT_32 requires unsigned comparison instead of the
+   * natural signed one. Use {@link #compareToMin(Comparable)} or the comparator returned by {@link #comparator()} to
+   * always get the proper ordering.
+   */
   abstract public T genericGetMin();
+
+  /**
+   * Returns the max value in the statistics. The java natural order of the returned type defined by {@link
+   * T#compareTo(Object)} might not be the proper one. For example, UINT_32 requires unsigned comparison instead of the
+   * natural signed one. Use {@link #compareToMax(Comparable)} or the comparator returned by {@link #comparator()} to
+   * always get the proper ordering.
+   */
   abstract public T genericGetMax();
 
+  /**
+   * Returns the comparator to be used to compare two generic values in the proper way (for example, unsigned comparison
+   * for UINT_32).
+   */
+  public final PrimitiveComparator<T> comparator() {
+    return comparator;
+  }
+
+  /**
+   * Compares the specified value to min in the proper way.
+   *
+   * @see Comparable#compareTo(Object)
```

Comment:
Because we don't want people to use `compareTo`, I don't think it makes sense to link to it here. You could link to `Comparator` instead or copy the description of the return values.